### DISCUSSION TITLE
Feature requests

### DISCUSSION THREAD

**Discussion by JustAMe56** [KevinBytesTheDust/CVAmp#214] [General]
First of all I love your program. I have been using it for a month and have a few quality of life suggestions:

1) spawning X amount of viewer with Y amount of interval

Example: Set max viewer count to 40, spawn one viewer each 30 seconds

I noticed that spawning a couple of viewers at the same time results in timeouts due to cpu overload.  But spawning them one by one with a few minutes  interval works great. Even on a 10 years lower end machine I manage to run 40+ viewers easily by spacing the time they spawn. Once the viewer is up and running ("green mode") it barely uses any resources. This would greatly increase the amount of viewer a computer can simulate without having to do some difficult optimization's.

I have been doing this manual now but it would be great to have it automated and i think it could benefit lots of users. 

2) Supporting rotating proxies from webshare. 

3) Remembering the URL so you don't have to write it each time. Or let us safe it and pick from a drop down box. 

Thank you very much

**Comment by jlplenio** [comment]
Thank you for your extensive feedback.

1. I refrain to providing this option because I want to prevent automation and commercialization of the software. I was already notified by ko-fi supportes, that this software in both the ko-fi and github version are renamed and sold on the WWW 👎 
This is enforced by users having to manually spawn and supervise the instances - I find this a good compromise.

2. I don't think this is possible, see here https://github.com/jlplenio/crude-twitch-viewer-bot/discussions/211#discussioncomment-7207891

3. I've currently avoided providing users with a configuration tab or file. This also where the URL could be saved. My reasoning is that if each user has varying and potentially conflicting settings, troubleshooting general issues could become exponentially more challenging. However, I do have plans to introduce a settings tab with persistent configurations and/or a configuration file in the future, definitely.

**Comment by Holli82** [comment]
> First of all I love your program. I have been using it for a month and have a few quality of life suggestions:
> 
>     1. spawning X amount of viewer with Y amount of interval
> 
> 
> Example: Set max viewer count to 40, spawn one viewer each 30 seconds
> 
> I noticed that spawning a couple of viewers at the same time results in timeouts due to cpu overload. But spawning them one by one with a few minutes interval works great. Even on a 10 years lower end machine I manage to run 40+ viewers easily by spacing the time they spawn. Once the viewer is up and running ("green mode") it barely uses any resources. This would greatly increase the amount of viewer a computer can simulate without having to do some difficult optimization's.
> 
> I have been doing this manual now but it would be great to have it automated and i think it could benefit lots of users.

@JustAMe56 You can realise this when you have a little Macro Recorder Running. Make this "Process" One Time and set a Loop Value for how many Viewers you want. (You need enougth Proxys) 

Example: 
I Start Macro Record , than Click Spawn Instance wait 30-90 Seconds how long it Takes to have it Green. 
Than Stop the Record set a Loop Value (How many Times it Loops) and than you have the Amount of Viewers you want. 

But dont forgget you need the Numbers of Proxys to how Many Viewers you want some of Reserve is not bad. :)

**Reply by jlplenio** [reply]
You sneaky boy, you. Happy holidays.

**Reply by Holli82** [reply]
Thanks wish you too a well Christmas Time.
It was a bit strange Situation, because why i answer on this Topic.
was thinking yesterday... mhh that were a great Feature :D for this helpful little application. Than i realise that were a Topic once the same Idea. So i let my Brain bit Working and has remember on a little Macro to dont get Kick because AFK Things. :D  

Its just a more Dirty way but with a little Macro Recording. But so you dont need to Program some extra in your code for it and its not self automated. So the User need to invest bit own Time to make it Working. I find its a good Solution for all who have interess to use it more automated.
